go-controller/pkg/ovnwebhook/podadmission.go

```diff
@@ -0,0 +1,167 @@
+package ovnwebhook
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/exp/maps"
+
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	listers "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kubevirt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// checkPodAnnot defines an additional checks for the allowed annotations
+type checkPodAnnot func(nodeLister listers.NodeLister, v annotationChange, pod *corev1.Pod, nodeName string) error
+
+// interconnectPodAnnotationChecks holds annotations allowed for ovnkube-node:<nodeName> users in IC environments
+var interconnectPodAnnotations = map[string]checkPodAnnot{
+	util.OvnPodAnnotationName: func(nodeLister listers.NodeLister, v annotationChange, pod *corev1.Pod, nodeName string) error {
+		// Ignore kubevirt pods with live migration, the IP can cross node-subnet boundaries
+		if kubevirt.IsPodLiveMigratable(pod) {
```

Comment:
Then this check could be easily circumvented by making us believe this is a kubevirt pod I guess. 

@qinqon @kyrtapz could we reject any update adding the label we are checking for?

Reply:
`kubevirt.io/allow-pod-bridge-network-live-migration` is not an annotation that ovnkube-node can set with the webhook in place, so it would have to be done by a different user. The changes introduced in this PR try to mostly limit what an ovnkube-node user can do to a cluster, especially workloads on different nodes. On the other hand if there is a straight forward way we could address the issue I am all for it.

Reply:
@jcaamano we can add an extra check for pod owner, it has to be the VirtualMachineInstance and since it has to be at the same namespace it means the malicious issuer will need to also have access to the VMI namespace, what do you think ?

Reply:
I guess that a malicious user could set the annotation, and then use the compromised ovnkube-node to change the pod ip to another workload ip or/and on another nodes subnet, and from there,  there might be trouble?

@qinqon with your option above what would prevent the malicious user to set the owner that it needs so that that check is bypassed? Does KAS make any check that an owner reference is valid? Is the VMI namespace protected and well known?

Reply:
We had an offline discussion about this.

Our first opinion is that we might not be concerned about this since the namespace where the live migration pods run should be owned by a cluster privileged user, so not owned by tenants.

If we should be concerned about this, we considered cross-checking that, if the pod is a kubevirt live migratable pod, and if the ip being set does not correspond with the node subnet, we could check that the ip comes from another live migratable pod. This though has performance consideration as we would need to bring in a pod informer caching all live migratable pods at least. 

The preferred option is not to be concerned about this.